### PR TITLE
Flatten the stdlib

### PR DESCRIPTION
**Pull request by nfurfaro**
While a final design for the organization of `std` will likely involve more discussion (& consideration of the proposed `Contract` type with some of the existing free functions converted to methods on the `Contract` type), the changes I've made here remove some weirdness.
This is a breaking change, so it might be a good time to clean at least this up.
I've left the `vm` directory as currently, it seems like the only good use of a subdirectory in `std`.
If anyone thinks this needs more discussion we can mark this a draft.

ref #1759

### CODE REVIEW

File: sway-lib-std/src/context.sw [hunk 1]
```diff
@@ -1,11 +1,11 @@
 //! Functionality for accessing context-specific information about the current contract or message.
 library context;
-dep context/registers;
-dep context/call_frames;
+dep registers;
+dep call_frames;
```
Comment (mohammadfawaz) [inline sway-lib-std/src/context.sw:4]: Are those still needed?

**Review by mohammadfawaz — COMMENTED** [review]
I'm generally in favour of this. It's currently a bit cumbersome to have to import `std::auth::chain::msg_sender()` just to access a utility that is so commonly used.

File: sway-lib-std/src/context.sw (hunk 1 above)
Comment (nfurfaro) [reply sway-lib-std/src/context.sw:4]: I thought so, but they are not. Will remove and update a few more paths.

**Review by mohammadfawaz — APPROVED** [review]
Awesome, :shipit:

**Review by Braqzen — APPROVED** [review]

**Pull request merged**